ios/RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -117,23 +116,13 @@ - (void)layoutSubviews
 
 - (void)setAnnotations:(NSMutableArray *)annotations
 {
-    _newAnnotations = annotations;
-    [self performSelector:@selector(updateAnnotations) withObject:nil afterDelay:0.5];
+    [self performSelector:@selector(updateAnnotations:) withObject:annotations afterDelay:0.5];
```

Comment:
That suggests that `-setAnnotations:` itself should perhaps be called on a delay. If you later call `-setAnnotations:`, a delay probably wouldn’t make sense – unless you’re also resetting the style on the same iteration of the run loop?